### ISSUE TITLE
upspin-deploy-gcp: fails when google cloud is apt-installed

### ISSUE THREAD

**Issue by rogpeppe** [upspin/upspin#454]
One of the first things that upspin-deploy-gcp does is to run "gcloud components install beta".
This fails when google cloud tools have been installed with apt:

	% upspin-deploy-gcp -create -domain upspin.rogpeppe.com -project rogpeppe-upspin
	upspin-deploy-gcp: Creating cloud services: Project="rogpeppe-upspin" Zone="us-central1-a" Prefix=""
	upspin-deploy-gcp: create: enableAPIs: installing gcloud beta components: exit status 1
	ERROR: (gcloud.components.install) 
	You cannot perform this action because the Cloud SDK component manager 
	is disabled for this installation. You can run the following command 
	to achieve the same result for this installation: 
	
	sudo apt-get install google-cloud-sdk
	
Note that google-cloud-sdk has already been installed.
When the relevant lines from Config.enableAPIs are removed, it works.

**Comment by rogpeppe** [comment]
Having said "it works", it did in fact fail. I guess the failure could be related:

Here was the output of upspin-deploy-gcp:

	% upspin-deploy-gcp -create -domain upspin.rogpeppe.com -project rogpeppe-upspin
	upspin-deploy-gcp: Creating cloud services: Project="rogpeppe-upspin" Zone="us-central1-a" Prefix=""
	upspin-deploy-gcp: Enabling API "cloudbuild.googleapis.com"
	upspin-deploy-gcp: Enabling API "compute_component"
	upspin-deploy-gcp: Enabling API "container"
	upspin-deploy-gcp: Enabling API "dns.googleapis.com"
	upspin-deploy-gcp: Enabling API "storage_api"
	upspin-deploy-gcp: Creating disk "dirserver"
	upspin-deploy-gcp: Creating network "network"
	upspin-deploy-gcp: Creating bucket "rogpeppe-upspin-cdbuild"
	upspin-deploy-gcp: Creating address "dir"
	upspin-deploy-gcp: Creating bucket "rogpeppe-upspin-letsencrypt"
	upspin-deploy-gcp: Creating bucket "rogpeppe-upspin-store"
	upspin-deploy-gcp: Building cloudbuild Docker image
	upspin-deploy-gcp: Pushing code to gs://rogpeppe-upspin-cdbuild/build/cloudbuild-def5d142a85942f482f5cb1a00c6694d.tar.gz
	upspin-deploy-gcp: Creating address "store"
	upspin-deploy-gcp: Logs at https://console.cloud.google.com/m/cloudstorage/b/rogpeppe-upspin-cdbuild/o/log-613dc4d2-2d11-47a6-a367-4594dd8547b1.txt
	upspin-deploy-gcp: Creating cluster "cluster"
	upspin-deploy-gcp: Build status: SUCCESS
	upspin-deploy-gcp: Cleaned up.
	
	upspin-deploy-gcp: Deploying Upspin servers [dirserver storeserver]: Project="rogpeppe-upspin" Zone="us-central1-a" Prefix=""
	upspin-deploy-gcp: Fetching kubectl credentials
	Fetching cluster endpoint and auth data.
	kubeconfig entry generated for cluster.
	upspin-deploy-gcp: Building "storeserver"
	upspin-deploy-gcp: Building "dirserver"
	upspin-deploy-gcp: deploy: dirserver: error copying "public.upspinkey" for dirserver: open /home/rog/upspin/deploy/upspin.rogpeppe.com/dirserver/public.upspinkey: no such file or directory
	% % upspin-deploy-gcp -create -domain upspin.rogpeppe.uk -project rogpeppe-upspin
	upspin-deploy-gcp: Creating cloud services: Project="rogpeppe-upspin" Zone="us-central1-a" Prefix=""
	upspin-deploy-gcp: Enabling API "compute_component"
	upspin-deploy-gcp: Enabling API "container"
	upspin-deploy-gcp: Enabling API "storage_api"
	upspin-deploy-gcp: Creating disk "dirserver"
	upspin-deploy-gcp: Creating network "network"
	upspin-deploy-gcp: Creating address "dir"
	upspin-deploy-gcp: Creating bucket "rogpeppe-upspin-cdbuild"
	upspin-deploy-gcp: Creating cluster "cluster"
	upspin-deploy-gcp: Creating address "store"
	upspin-deploy-gcp: Creating bucket "rogpeppe-upspin-letsencrypt"
	upspin-deploy-gcp: Creating bucket "rogpeppe-upspin-store"
	upspin-deploy-gcp: Building cloudbuild Docker image
	upspin-deploy-gcp: Pushing code to gs://rogpeppe-upspin-cdbuild/build/cloudbuild-710e90e59da9142c0f68217edec87259.tar.gz
	upspin-deploy-gcp: Logs at https://console.cloud.google.com/m/cloudstorage/b/rogpeppe-upspin-cdbuild/o/log-3af42f5b-efa4-4bfc-9e41-8e718b219f3d.txt
	upspin-deploy-gcp: Build status: SUCCESS
	upspin-deploy-gcp: Cleaned up.
	upspin-deploy-gcp: Deploying Upspin servers [dirserver storeserver]: Project="rogpeppe-upspin" Zone="us-central1-a" Prefix=""
	upspin-deploy-gcp: Fetching kubectl credentials
	Fetching cluster endpoint and auth data.
	kubeconfig entry generated for cluster.
	upspin-deploy-gcp: Building "storeserver"
	upspin-deploy-gcp: Building "dirserver"
	upspin-deploy-gcp: Building Docker image "dirserver"
	upspin-deploy-gcp: Pushing code to gs://rogpeppe-upspin-cdbuild/build/dirserver-fa2928a28f44edb1f2596ab4ff053fd3.tar.gz
	upspin-deploy-gcp: Building Docker image "storeserver"
	upspin-deploy-gcp: Pushing code to gs://rogpeppe-upspin-cdbuild/build/storeserver-49cc0714ae1a002575ff802c749f371d.tar.gz
	upspin-deploy-gcp: Logs at https://console.cloud.google.com/m/cloudstorage/b/rogpeppe-upspin-cdbuild/o/log-96d4b7ce-f61c-471c-b534-d3a8ca2c45d5.txt
	upspin-deploy-gcp: Logs at https://console.cloud.google.com/m/cloudstorage/b/rogpeppe-upspin-cdbuild/o/log-9ffc2752-7ebc-4879-b9a0-6886c220e0e1.txt
	upspin-deploy-gcp: Build status: FAILURE
	upspin-deploy-gcp: Cleaned up.
	upspin-deploy-gcp: deploy: storeserver: cdbuild failed

	starting build "96d4b7ce-f61c-471c-b534-d3a8ca2c45d5"

Here are the contents of one of the linked server logs (both contain the same errors):

	FETCHSOURCE
	Fetching storage object: gs://rogpeppe-upspin-cdbuild/build/dirserver-fa2928a28f44edb1f2596ab4ff053fd3.tar.gz#1504169367783188
	Copying gs://rogpeppe-upspin-cdbuild/build/dirserver-fa2928a28f44edb1f2596ab4ff053fd3.tar.gz#1504169367783188...
	/ [0 files][    0.0 B/  1.8 MiB]                                                
	-
	- [1 files][  1.8 MiB/  1.8 MiB]                                                
	Operation completed over 1 objects/1.8 MiB.                                      
	BUILD
	Step #0: Pulling image: gcr.io/rogpeppe-upspin/cloudbuild
	Step #0: Using default tag: latest
	Step #0: latest: Pulling from rogpeppe-upspin/cloudbuild
	Step #0: ad74af05f5a2: Already exists
	Step #0: 2b032b8bbe8b: Already exists
	Step #0: a9a5b35f6ead: Already exists
	Step #0: 098d41be9c78: Pulling fs layer
	Step #0: 334cc5cf9a4f: Pulling fs layer
	Step #0: ea4f828f5c02: Pulling fs layer
	Step #0: 96051881f39c: Pulling fs layer
	Step #0: 96051881f39c: Waiting
	Step #0: ea4f828f5c02: Verifying Checksum
	Step #0: ea4f828f5c02: Download complete
	Step #0: 96051881f39c: Verifying Checksum
	Step #0: 96051881f39c: Download complete
	Step #0: 334cc5cf9a4f: Verifying Checksum
	Step #0: 334cc5cf9a4f: Download complete
	Step #0: 098d41be9c78: Verifying Checksum
	Step #0: 098d41be9c78: Download complete
	Step #0: 098d41be9c78: Pull complete
	Step #0: 334cc5cf9a4f: Pull complete
	Step #0: ea4f828f5c02: Pull complete
	Step #0: 96051881f39c: Pull complete
	Step #0: Digest: sha256:19a6b3c82b7060e10626b13423dbd459750e15f9fd341de91d00168a61ce6ac1
	Step #0: Status: Downloaded newer image for gcr.io/rogpeppe-upspin/cloudbuild:latest
	Starting Step #0
	Step #0: # cloud.google.com/go/logging/apiv2
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:125: undefined: grpc.Version
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:126: undefined: gax.XGoogHeader
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:160: cannot use func literal (type func("golang.org/x/net/context".Context, gax.CallSettings) error) as type gax.APICall in argument to gax.Invoke
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:162: settings.GRPC undefined (type gax.CallSettings has no field or method GRPC)
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:187: cannot use func literal (type func("golang.org/x/net/context".Context, gax.CallSettings) error) as type gax.APICall in argument to gax.Invoke
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:189: settings.GRPC undefined (type gax.CallSettings has no field or method GRPC)
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:207: cannot use func literal (type func("golang.org/x/net/context".Context, gax.CallSettings) error) as type gax.APICall in argument to gax.Invoke
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:209: settings.GRPC undefined (type gax.CallSettings has no field or method GRPC)
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:230: cannot use func literal (type func("golang.org/x/net/context".Context, gax.CallSettings) error) as type gax.APICall in argument to gax.Invoke
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:232: settings.GRPC undefined (type gax.CallSettings has no field or method GRPC)
	Step #0: src/cloud.google.com/go/logging/apiv2/config_client.go:230: too many errors
	Finished Step #0
	ERROR
	ERROR: build step "gcr.io/rogpeppe-upspin/cloudbuild" failed: exit status 2

**Comment by adg** [comment]
I strongly advise against using `upspin-deploy-gcp`. We use it for upspin.io but it's not something we want to continue to support for general use. There are too many moving parts.

**Comment by rogpeppe** [comment]
OK. What *is* the recommended easiest way of deploying an upspin server in the cloud then?

**Comment by robpike** [comment]
I used the instructions at upspin.io at https://upspin.io/doc/server_setup.md and https://upspin.io/doc/server_setup_ubuntu.md to set up my own upspinserver on GCP, but I know too much and might have avoided a problem you hit. I am using Mac, though, and so have never used apt (or homebrew for that matter; long story). I just got the SDK from the public Google cloud site.

Without further information, my only suggestion is to avoid apt and/or run go get -u on everything. Or perhaps if I did go get -u on everything I'd hit this; bugs do get deployed and I haven't updated for a couple of weeks.
